File: tests/xcalablemp/global-view/intrinsic/C-Round/chk/chk.c
#include <stdio.h>
#include <stdlib.h>
#include "xmp.h"

#pragma xmp nodes p(*)

int chk_int(int ierr){

#pragma xmp reduction (max:ierr)
#pragma xmp task on p(1)
{
  if ( ierr == 0 ) {
     printf("PASS\n");
  }else{
     printf("ERROR\n");
     exit(1);
  }
}
  return 0;

}

int chk_int2(int ierr){

#pragma xmp reduction (max:ierr)
  int irank=xmp_node_num();

  if(irank==1){
    if ( ierr == 0 ) {
       printf("PASS\n");
    }else{
       printf("ERROR\n");
       exit(1);
    }
  }
  return 0;
}

File: tests/XMP/global-view/intrinsic/F/chk/chk.f90
subroutine chk_int(ierr)
integer ierr
!$xmp nodes p(*)

!$xmp task on p(1)
  if ( ierr .eq. 0 ) then
     write(*,*) "PASS"
  else
     write(*,*) "ERROR"
     call exit(1)
  endif
!$xmp end task

return
end subroutine

subroutine chk_int2(ierr)
integer irank, xmp_node_num, ierr

!$xmp reduction (max:ierr)
  irank=xmp_node_num()
  if ( irank .eq. 1)then
    if ( ierr .eq. 0 ) then
       write(*,*) "PASS"
    else
       write(*,*) "ERROR"
       call exit(1)
    endif
  endif
return
end subroutine

subroutine chk_int3(ierr, n)
integer ierr
!$xmp nodes p(*)

!$xmp task on p(n)
  if ( ierr .eq. 0 ) then
     write(*,*) "PASS"
  else
     write(*,*) "ERROR"
     call exit(1)
  endif
!$xmp end task

return
end subroutine
